frontend/src/Troop/index.jsx

```diff
@@ -13,21 +19,63 @@ const styles = StyleSheet.create({
   },
 });
 
-function TroopLevels() {
+function TroopLevels({ level1TroopNum, level2TroopNum, level3TroopNum }) {
   return (
     <View style={styles.levelList}>
-      <TroopLevel level={1} troops={13} />
-      <TroopLevel level={2} troops={8} />
-      <TroopLevel level={3} troops={2} />
+      {level1TroopNum !== 0 && <TroopLevel level={1} troops={level1TroopNum} />}
+      {level2TroopNum !== 0 && <TroopLevel level={2} troops={level2TroopNum} />}
+      {level3TroopNum !== 0 && <TroopLevel level={3} troops={level3TroopNum} />}
     </View>
   );
 }
 
+TroopLevels.propTypes = {
+  level1TroopNum: PropTypes.number.isRequired,
+  level2TroopNum: PropTypes.number.isRequired,
+  level3TroopNum: PropTypes.number.isRequired,
+};
+
 function Troop() {
+  const troops = useSelector((state) => state.troop.troops);
+  const isLoading = useSelector((state) => state.troop.isLoading);
+  const fetchError = useSelector((state) => state.troop.fetchError);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(fetchTroops());
+  }, []);
+
+  if (fetchError !== null) {
+    return (
+      <Text>{fetchError.message}</Text>
+    );
+  }
+
+  if (isLoading) {
+    return (
+      <ActivityIndicator size="large" color="#10978b" />
+    );
+  }
```

Comment:
put this under the calculations please

Reply:
The calculation result won't render if isloading. Is there any reason for this?